### ISSUE TITLE
Allow custom path specfication in S3_ALIAS_HOST

### ISSUE THREAD

**Issue by Sir-Boops** [mastodon/mastodon#9853]
Right now when using `S3_ALIAS_HOST` you only can set the domain then the masto filename is appended to that what I'd like to see happen is allowing someone to set a domain and/or path then having the masto filename appended to that!

Or perhaps a new option like `S3_ALIAS_HOST_PATH` ?

This would allow proxying services like b2 directly through cloudflare allowing admins to make usage of the bandwidth alliance! :)

**Comment by benpro** [comment]
Yup, actually now that B2 support S3 API it is really an important feature I think.

Example URL:

**Type A**: https://toots-benpro-fr.s3.eu-central-003.backblazeb2.com/accounts/avatars/000/000/001/original/4a9e11478e1f4fce.png
**Type B**: https://s3.eu-central-003.backblazeb2.com/toots-benpro-fr/accounts/avatars/000/000/001/original/4a9e11478e1f4fce.png

But if you use CloudFlare in front of B2 S3 API, you cannot use the S3 API with the **type A** URL nor **type B**. You need another URL format...

**Type C**: https://toots-benpro-fr.benpro.download/file/toots-benpro-fr/accounts/avatars/000/000/001/original/4a9e11478e1f4fce.png **which is a `S3_ALIAS_HOST` with a specific path**: https://toots-benpro-fr.benpro.download/file/toots-benpro-fr/
Basically `$customDomainOnCloudFlare/file/$B2BucketName/`

**Comment by supernovae** [comment]
Is there any way to use a URL instead of host here? I'd like to add the bucket to the path - if its not via hostname is there another way to add this that i'm not aware of?

I'd really like to use cloudflare's new bandwidth alliance partners and set up the ACLS so they have direct s3 object read access and i won't get penalized/kicked off for too much cache

**Comment by blastehh** [comment]
You can use a URL transform rule directly on cloudflare to make use of their bucket:
![image](https://user-images.githubusercontent.com/8324677/202920179-3ffbb981-8996-47ce-92b2-adb7e92a8a4b.png)

S3_ALIAS_HOST=static.domain.com

and then cname to the b2 friendly url endpoint
![image](https://user-images.githubusercontent.com/8324677/202920207-5720361c-a89c-4fbf-8106-74b1d66b6b08.png)

**Comment by eric-pierce** [comment]
@blastehh  This is what I ended up doing, but it does require Cloudflare. Here's a guide Backblaze put together about removing unnecessary headers, and optimizing cache

**Comment by nightpool** [comment]
Looks like this was resolved with Cloudflare Transform Rules
